src/util/mempool.h

```diff
@@ -28,19 +28,8 @@ typedef struct {
   mempool_free_fn free;
   size_t initialCap;  // Initial size of the pool
   size_t maxCap;      // maxmimum size of the pool
-
-  /**
-   * if true, will be added to the list of global mempool objects which
-   * will be destroyed via mempool_free_global(). This also means you
-   * cannot call mempool_destroy() on it manually
-   */
-  int isGlobal;
```

Comment:
instead of passing the global flag, pass the parameters to `mempool_test_set_global` instead of `mempool_new` (which you can use for non-global mempools)